File: 卡码/代码随想录/C++/二叉树/03-二叉树修改与构造/LeetCode_0226.cpp
//226. ·­×ª¶ş²æÊ÷
#include <iostream>
#include "../TreeNode.h"
using namespace std;

class Solution {
public:
  TreeNode* invertTree(TreeNode* root) {
		if(!root) return root;
		swap(root->left,root->right);
		invertTree(root->left);
		invertTree(root->right);
		return root;
  }
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/Java/codeThoughts/src/main/java/com/kurumi/leetCode/codeThoughts/二叉树/_03_二叉树修改与构造/LeetCode_0226.java
package com.kurumi.leetCode.codeThoughts.二叉树._03_二叉树修改与构造;

import com.kurumi.leetCode.codeThoughts.entity.TreeNode;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 226. 翻转二叉树
 */
public class LeetCode_0226 {
    static class Solution {
        public TreeNode invertTree(TreeNode root) {
            if (root == null)
                return root;
            swap(root);
            invertTree(root.left);
            invertTree(root.right);
            return root;
        }

        public void swap(TreeNode root) {
            TreeNode tmp = root.left;
            root.left = root.right;
            root.right = tmp;
        }
    }
}
